addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -75,14 +75,14 @@ def _validate_phone_number(self, phone):
         :raise ValidationError: If the phone number is missing or incorrect.
         """
         if not phone:
-            raise ValidationError("Razorpay: " + _("The phone number is missing."))
+            return ""
```

Comment:
Did you check passing empty string as contact `'contact': self._validate_phone_number(self.partner_phone),` does not cause any issue? :thinking: 
Same when requesting recurring payment in `def _send_payment_request(self):` method. Is it working properly?

Reply:
for normal payment, it does not need a phone.
but for recurring it's required

Reply:
As we are using same flow for both type of payments
I don't think we need to add `if...else` block as error or phone number should be same for both flows
Additionally no need to add `try...exception` block as you can just remove `raise_exception` argument so this way you don't need to check `if not phone` here
```python
 phone = self._phone_format(number=phone, country=self.partner_country_id) or ""
``` 
Did you have runbot or ngrok URL where I can test it is not giving any issue?

Reply:
My bad :sweat_smile: , should have better check it before making a quick push

Feel free to checkout in my branch once I m done with the changes :wink: